pkg/storage/secret/manager/manager.go

```diff
@@ -0,0 +1,175 @@
+package manager
+
+import (
+	"cmp"
+	"context"
+	"fmt"
+
+	secret "github.com/grafana/grafana/pkg/apis/secret/v0alpha1"
+	"github.com/grafana/grafana/pkg/storage/secret/keeper"
+	"github.com/grafana/grafana/pkg/storage/secret/metastorage"
+)
+
+// This is the entrypoint for a securevalue.
+// Here it gets encrypted and decrypted.
+// Here it gets stored and retrieved.
+// Many strategies available, local and remote for each.
+type SecretsManagerService struct {
+	metadataStore metastorage.SecretMetadataStore
+	localKeeper   keeper.SecretKeeperService
+	remoteKeepers map[keeper.SecretKeeperRemoteProvider]keeper.SecretKeeperService
+}
+
+// TODO: initialization of this thing.
+func NewSecretsManagerService(
+	metadataStore metastorage.SecretMetadataStore,
+	localKeeper keeper.SecretKeeperService,
+	remoteKeepers map[keeper.SecretKeeperRemoteProvider]keeper.SecretKeeperService,
+) (*SecretsManagerService, error) {
+	return &SecretsManagerService{metadataStore, localKeeper, remoteKeepers}, nil
+}
+
+// Create will encrypt a plaintext value and store it in a vault storage.
+// TODO: what to return apart from the error?
+func (s *SecretsManagerService) Create(ctx context.Context, securevalue *secret.SecureValue) (any, error) {
+	// 1. Just a value is provided, use the local keeper to encrypt and store in SQL.
+	if securevalue.Spec.Value != "" && securevalue.Spec.Manager == "" {
+		ciphertext, err := s.localKeeper.Encryption().Encrypt(ctx, []byte(securevalue.Spec.Value))
```

Comment:
in my envisioning of the architecture, this logic occurs down a level when Store() is called on the keeper.

Reply:
@mmandrus How would the use-case where you want a securevalue encrypted remotely (i.e. AWS KMS, could be any) but stored in SQL work in this case?